mdal/mdal_utils.cpp

```diff
@@ -796,3 +796,70 @@ void MDAL::Error::setDriver( std::string driverName )
 {
   driver = driverName;
 }
+
+void parseDriverFromUri( const std::string &uri, std::string &driver )
+{
+  bool hasDriverSet = ( uri.find( ":\"" ) != std::string::npos );
+  driver = "";
+
+  if ( !hasDriverSet )
+    return;
+
+  driver = MDAL::split( uri, ":\"" )[0];
```

Comment:
cannot you do split and if the resulting array has length 0 return "" otherwise return array[0] ?